solidity/contracts/ValidatorRegistration.sol

```diff
@@ -1,23 +1,113 @@
-pragma solidity ^0.4.23;
+// Adapted from: https://github.com/prysmaticlabs/prysm/blob/master/contracts/validator-registration-contract/validator_registration.sol
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
 
 contract ValidatorRegistration {
 
-  event Deposit(bytes32 _pubkey, uint256 _withdrawal_shard_id, address _withdrawal_address, bytes32 _randao_commitment);
+    event Eth1Deposit(
+        bytes previousReceiptRoot,
+        bytes data,
+        uint totalDepositcount
+    );
 
-  mapping(bytes32 => bool) public used_pubkeys;
+    event ChainStart(
+        bytes receiptRoot,
+        bytes time
+    );
```

Comment:
Perhaps I'm naive with solidity, but I do believe the events should follow the same convention in the vyper code. Specifically, `totalDepositcount` should be `deposit_count`, otherwise when reading the event logs, our chain would test against a different value then the official contract that is released on the eth1.x?

Reply:
@GregTheGreek I believe it mostly depends on what language the deployed contract is written in. If it's in Solidity, it's very likely that the names of the event arguments are in camelCase, but if it's written in Vyper it's likely that they're in snake_case. From a client perpective, the changes in format should be trivial, so I think it's necessary to follow camelCase for now.

Also, this can be brought up in an implementor's call, as it's a very good question!

Reply:
I do believe it is the keccak of the function parameter names that are used to produce the bloom filter for the logs, but we should definitely double check with someone who knows vyper

Reply:
My issue was surrounding the logs themselves, but they are returned as `result: { '0':... , '1': ..., ...}` so we're good.

@ansermino I don't think thats an issue tbh. From what i found, the param names within events is a placeholder and isn't used in the computation to save the tx_receipts.

Reply:
@GregTheGreek true! its just the event name that gets hashed, the values are referenced by indexes (https://ethereum.github.io/yellowpaper/paper.pdf equation 24)

Reply:
I'm going to resolve this.